examples/performance/iterator-offsets/iterator-offsets.cxx

```diff
@@ -0,0 +1,291 @@
+  Options *globalOptions = Options::getRoot();
+  Options *modelOpts = globalOptions->getSection("performanceIterator");
+  int NUM_LOOPS;
+  OPTION(modelOpts, NUM_LOOPS, 100);
+  bool profileMode, includeHeader;
+  OPTION(modelOpts, profileMode, false);
+  OPTION(modelOpts, includeHeader, false);
+
+  ConditionalOutput time_output{Output::getInstance()};
+  time_output.enable(true);
+
+  const Field3D a{1.0};
+  const Field3D b{2.0};
+
+  Field3D result;
+  result.allocate();
+
+  const int len = mesh->LocalNx*mesh->LocalNy*mesh->LocalNz;
+
+  // Nested loops over block data
+  ITERATOR_TEST_BLOCK(
+    "Nested loop",
+    for(int i=0;i<mesh->LocalNx;++i) {
+      for(int j=mesh->ystart;j<mesh->yend;++j) {
+        for(int k=0;k<mesh->LocalNz;++k) {
+          result(i,j,k) = (a(i,j+1,k) - a(i,j-1,k));
+        }
+      }
+    }
+    );
+
+#ifdef _OPENMP
+  ITERATOR_TEST_BLOCK(
+    "Nested loop (omp)",
+    BOUT_OMP(parallel for)
+    for(int i=0;i<mesh->LocalNx;++i) {
+      for(int j=mesh->ystart;j<mesh->yend;++j) {
+        for(int k=0;k<mesh->LocalNz;++k) {
+          result(i,j,k) = (a(i,j+1,k) - a(i,j-1,k));
+        }
+      }
+    }
+    );
+#endif
+
+  // Range based for DataIterator with indices
+  ITERATOR_TEST_BLOCK(
+    "C++11 range-based for (omp)",
+    BOUT_OMP(parallel)
+    for(auto i : result.region(RGN_NOY)){
+      result(i.x,i.y,i.z) = (a(i.x,i.y+1,i.z) - a(i.x,i.y-1,i.z));
+    }
+    );
+
+  // Range based DataIterator
+  ITERATOR_TEST_BLOCK(
+    "C++11 range-based for [i] (omp)",
+    BOUT_OMP(parallel)
+    for(const auto &i : result.region(RGN_NOY)){
+      result[i] = (a[i.yp()] - a[i.ym()]);
+    }
+    );
+
+  ITERATOR_TEST_BLOCK(
+    "C++11 range-based for over Region [i] (omp)",
+    BOUT_OMP(parallel)
+    {
+      for(const auto &i : mesh->getRegion3D("RGN_NOY")){
+        result[i] = (a[i.yp()] - a[i.ym()]);
+      }
+    }
+    );
+
+#ifdef _OPENMP
+  ITERATOR_TEST_BLOCK(
+    "C++11 range-based for [i] with stencil (omp)",
+    BOUT_OMP(parallel)
+    {
+      stencil s;
+      for(const auto &i : result.region(RGN_NOY)){
+        s.mm = nan("");
+        s.m = a[i.ym()];
+        s.c = a[i];
+        s.p = a[i.yp()];
+        s.pp = nan("");
+        result[i] = (s.p - s.m);
+      }
+    }
+    );
+#endif
+  ITERATOR_TEST_BLOCK(
+    "C++11 range-based for [i] with stencil (serial)",
+    stencil s;
+    for(const auto &i : result.region(RGN_NOY)){
+      s.mm = nan("");
+      s.m = a[i.ym()];
+      s.c = a[i];
+      s.p = a[i.yp()];
+      s.pp = nan("");
+      result[i] = (s.p - s.m);
+    }
+    );
+
+  // Region macro
+  ITERATOR_TEST_BLOCK(
+      "Region with stencil (serial)",
+      stencil s;
+      BLOCK_REGION_LOOP_SERIAL(mesh->getRegion3D("RGN_NOY"), i,
+        s.mm = nan("");
+        s.m = a[i.ym()];
+        s.c = a[i];
+        s.p = a[i.yp()];
+        s.pp = nan("");
+
+        result[i] = (s.p - s.m);
+        );
+    );
+#ifdef _OPENMP
+
+  ITERATOR_TEST_BLOCK(
```

Comment:
I think the distinction with the `(inside)` case below has disappeared due to the removal of the `IndexOffset` class. I think one can be removed and the name updated.